drivers/scheduler/k8s/k8s.go

```diff
@@ -7624,3 +7634,45 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	cloneStorageClass := originalStorageClass
+	clonedSCName := cloneStorageClass.Name + "-clone"
+	clonedSCNotFound := false
+
+	t := func() (interface{}, bool, error) {
+		_, err := k8sStorage.GetStorageClass(clonedSCName)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				clonedSCNotFound = true
+				return "", false, err
+			}
+			return "", true, err
+		}
+		return "", false, nil
+	}
+
+	_, err := task.DoRetryWithTimeout(t, 2*time.Minute, 10*time.Second)
+	if err != nil {
+		if clonedSCNotFound {
+			log.Infof("Cloned SC with the name %v does not exist and will be created", clonedSCName)
+
+			cloneStorageClass.Name = clonedSCName
+			immediate := storageapi.VolumeBindingImmediate
+			cloneStorageClass.VolumeBindingMode = &immediate
+			cloneStorageClass.ResourceVersion = ""
+			cloneStorageClass.UID = ""
+
+			clonedSCobj, err := k8sStorage.CreateStorageClass(cloneStorageClass)
+			if err != nil {
+				log.Infof("Error occurred while creating cloned storage class:- %v", err)
```

Comment:
Should this be a higher severity log like `log.Errorf` since it indicates a failure in the function's main task?

Reply:
Yes. Fixed it.